### PR TITLE
Update eslint rules and refactoring types.

### PR DESCRIPTION
**Pull request by miettal**
1. banned using Object, Function types
2. refactor .eslintrc

### CODE REVIEW

File: ui/.eslintrc.json [hunk 1]
```diff
@@ -30,15 +30,8 @@
...
+        "curly": "error",
         "unused-imports/no-unused-imports": "error",
-        "@stylistic/semi": [
-          "error",
-          "always"
-        ],
-        "@stylistic/quote-props": [
-          "warn",
-          "consistent"
-        ],
```
Comment (lukasrgr) [inline ui/.eslintrc.json:41]: could you fix formatting, why did you change the order around?

File: ui/src/app/shared/service/service.ts [hunk 2]
```diff
@@ -126,7 +126,7 @@ export class Service extends AbstractService {
     // this.notify(notification);
   }
 
-  public setCurrentComponent(currentPageTitle: string | { languageKey: string, interpolateParams?: Object }, activatedRoute: ActivatedRoute): Promise<Edge> {
+  public setCurrentComponent(currentPageTitle: string | { languageKey: string, interpolateParams?: {} }, activatedRoute: ActivatedRoute): Promise<Edge> {
```
Comment (lukasrgr) [inline ui/src/app/shared/service/service.ts:129]: ```{}``` is better than ```Object``` but still not good, anyway this method is deprecated and shouldnt be used in the future anyway

File: ui/.eslintrc.json [hunk 3]
```diff
@@ -90,9 +84,7 @@
           {
             "extendDefaults": true,
             "types": {
-              "{}": false,
-              "Object": false,
-              "Function": false
+              "{}": false
```
Comment (lukasrgr) [inline ui/.eslintrc.json:87]: i would ban empty object as well, otherwise, type safety is not guaranteed

**Review by lukasrgr — CHANGES_REQUESTED** [review]

File: ui/.eslintrc.json (hunk 1 above)
Comment (miettal) [reply ui/.eslintrc.json:41]: to clear rule cateogry. I commented on this commit.
https://github.com/OpenEMS/openems/pull/2676/commits/4a1365c43ba1826b06f235a300fb739284304ffe

File: ui/.eslintrc.json (hunk 3 above)
Comment (miettal) [reply ui/.eslintrc.json:87]: Yes I want to ban {}. but in this repo there are using many {} type. especially {} used in json rpc related type. this means if I remove {} in this repo, this will be super heavy task(Many typing will needed....). so I focused to ban Object and Function.

Comment (miettal) [reply ui/.eslintrc.json:87]: Reducing `any` type and banning `{}`, applying strictNullChecks is very very important...
if I have time, I will address this kind little by little in other PR.

Comment (lukasrgr) [reply ui/.eslintrc.json:87]: i understand in terms of being a heavy task, so i approve

**Review by lukasrgr — APPROVED** [review]
thx for your contribution 🚀

**Review by sfeilmeier — APPROVED** [review]
